tuxemon/event/actions/translated_dialog.py

```diff
@@ -52,41 +37,25 @@ class TranslatedDialogAction(EventAction):
     """
 
     name = "translated_dialog"
-    raw_parameters: Sequence[str] = field(init=False)
-
-    def __init__(self, *args: Any) -> None:
-        super().__init__()
-        self.raw_parameters = args
+    raw_parameters: str
+    avatar: Union[int, str, None] = None
+    style: Optional[str] = None
 
     def start(self) -> None:
-        key = self.raw_parameters[0]
-        replace = []
+        key = process_translate_text(self.session, self.raw_parameters, [])
+
         avatar = None
-        for param in self.raw_parameters[1:]:
-            if "=" in param:
-                replace.append(param)
-            else:
-                avatar = get_avatar(self.session, param)
+        if self.avatar:
+            avatar = get_avatar(self.session, self.avatar)
```

Comment:
Just by curiosity, what's the difference between `avatar` and `self.avatar`?

If there is a difference, I think we should update the name of one of these identifiers to avoid confusion.

Reply:
done, it's more clear